tests/RobotFramework/tests/cumulocity/restart/restart_device_child.robot

```diff
@@ -0,0 +1,64 @@
+*** Settings ***
+Resource            ../../../resources/common.resource
+Library             Cumulocity
+Library             ThinEdgeIO
+
+Suite Setup         Custom Setup
+Test Teardown       Get Logs
+
+Test Tags           theme:c8y    theme:troubleshooting    theme:childdevices
+
+
+*** Test Cases ***
+Support restarting the device
+    Cumulocity.Should Contain Supported Operations    c8y_Restart
+    ${operation}=    Cumulocity.Restart Device
+    Operation Should Be SUCCESSFUL    ${operation}    timeout=180
+
+
+*** Keywords ***
+Setup Child Device
+    ThinEdgeIO.Set Device Context    ${CHILD_SN}
+    Execute Command    sudo dpkg -i packages/tedge_*.deb
+
+    Execute Command    sudo tedge config set mqtt.client.host ${PARENT_IP}
+    Execute Command    sudo tedge config set mqtt.client.port 1883
+    Execute Command    sudo tedge config set mqtt.topic_root te
+    Execute Command    sudo tedge config set mqtt.device_topic_id "device/${CHILD_SN}//"
+
+    # Install plugin after the default settings have been updated to prevent it from starting up as the main plugin
+    Execute Command    sudo dpkg -i packages/tedge-agent*.deb
+    Execute Command    sudo systemctl enable tedge-agent
+    Execute Command    sudo systemctl start tedge-agent
+
+    Transfer To Device    ${CURDIR}/*.sh    /usr/bin/
+    Transfer To Device    ${CURDIR}/*.service    /etc/systemd/system/
+    Execute Command    apt-get install -y systemd-sysv && chmod a+x /usr/bin/*.sh && chmod 644 /etc/systemd/system/*.service && systemctl enable on_startup.service
+    Execute Command    cmd=echo 'tedge ALL = (ALL) NOPASSWD: /usr/bin/tedge, /etc/tedge/sm-plugins/[a-zA-Z0-9]*, /bin/sync, /sbin/init, /sbin/shutdown, /usr/bin/on_shutdown.sh' > /etc/sudoers.d/tedge
+    Execute Command    cmd=sed -i 's|reboot =.*|reboot = ["/usr/bin/on_shutdown.sh"]|g' /etc/tedge/system.toml
+
+    # WORKAROUND: Uncomment next line once https://github.com/thin-edge/thin-edge.io/issues/2253 has been resolved
+    # ThinEdgeIO.Service Health Status Should Be Up    tedge-agent    device=${CHILD_SN}
+
+Custom Setup
+    # Parent
+    ${parent_sn}=    Setup    skip_bootstrap=${True}
+    Set Suite Variable    $PARENT_SN    ${parent_sn}
+    Execute Command           test -f ./bootstrap.sh && ./bootstrap.sh --no-connect || true
+
+    ${parent_ip}=    Get IP Address
+    Set Suite Variable    $PARENT_IP    ${parent_ip}
+    Execute Command    sudo tedge config set c8y.enable.log_management true
+    Execute Command    sudo tedge config set mqtt.external.bind.address ${PARENT_IP}
+    Execute Command    sudo tedge config set mqtt.external.bind.port 1883
+
+    ThinEdgeIO.Connect Mapper    c8y
+    ThinEdgeIO.Service Health Status Should Be Up    tedge-mapper-c8y
+
+    # Child
+    ${child_sn}=    Setup    skip_bootstrap=${True}
+    Set Suite Variable    $CHILD_SN    ${child_sn}
+    Register Child Device    parent_name=${PARENT_SN}    child_name=${CHILD_SN}    supported_operations=c8y_LogfileRequest
```

Comment:
Support of `c8y_LogfileRequest` is explicitly added here. Why? I would expect to have this added when `tedge_log_plugin` starts.

Reply:
Sorry that is my mistake (copy paste error). Maybe this is the source of the error. When I wrote the system tests for the tedge-log-plugin the supported operations mechanism was not implemented, so I compensated for it...I'll double check if this is the case now or not and update the tests accordingly.

Reply:
I've repeated the checks, and tedge does not forward the locally registered capabilities to Cumulocity...so this functionality probably hasn't been implemented yet.

Reply:
I'm missing something. I tested manual registration is working as I was expecting.

1. On start, `tedge-agent` publishes its capabilities `[te/device/child777-002///cmd/restart] {}`
2. On reception of this message, `c8y-mapper` publishes an auto-registration messages : `[te/device/child777-002//] { "@type":"child-device", "@id":"child777-002"}`.
3. On reception of this message, `c8y-mapper` also registers the child device and its capabilities:
     1. `[c8y/s/us] 101,child777-002,child777-002,thin-edge.io-child`
     2. `[c8y/s/us/child777-002] 114,c8y_Restart`

Reply:
Strange. I'll have another look, and try and strip back the test setup to a bare minimum, though I believe the test was already fairly minimal...

Reply:
> though I believe the test was already fairly minimal...

The test looks good. However, it's not obvious for me to see which checks are running on the main device and which on the child device.

Reply:
Ok, I've fixed it...the child device and capability are being registered correctly without doing any manual file creation (which what the test was previously doing).

So the only thing which is broken is the updating of the c8y_Restart operation status when operating on a child device.

Reply:
> Ok, I've fixed it...the child device and capability are being registered correctly without doing any manual file creation (which what the test was previously doing).

A file is created by the c8y mapper on the main device.

> So the only thing which is broken is the updating of the c8y_Restart operation status when operating on a child device.

Thanks. I will have a look.